src/ddsrt/include/dds/ddsrt/ifaddrs.h

```diff
@@ -24,41 +24,47 @@ extern "C" {
 
 /** @brief The interface type */
 enum ddsrt_iftype {
-  DDSRT_IFTYPE_UNKNOWN,
-  DDSRT_IFTYPE_WIRED,
-  DDSRT_IFTYPE_WIFI
+  DDSRT_IFTYPE_UNKNOWN, ///< An unknown interface
+  DDSRT_IFTYPE_WIRED, ///< A wired interface (e.g. ethernet)
+  DDSRT_IFTYPE_WIFI ///< A wifi interface
 };
 
 /** @brief Linked list of interface addresses */
 struct ddsrt_ifaddrs {
-  struct ddsrt_ifaddrs *next;
-  char *name;
-  uint32_t index;
-  uint32_t flags;
-  enum ddsrt_iftype type;
-  struct sockaddr *addr;
-  struct sockaddr *netmask;
-  struct sockaddr *broadaddr;
+  struct ddsrt_ifaddrs *next; ///< next pointer of the linked list
+  char *name; ///< name of the interface
+  uint32_t index; ///< the index of the interface
```

Comment:
The "index of the interface" is a really just a unique number associated with the interface by the operating system, it just happens to be called "index" historically, from what I can tell. The socket abstractions are much less well-documented than you might be expecting ...

So
```suggestion
  uint32_t index; ///< a unique integer associated with the interface
```
might be more useful to the average user